pykern/pkcli/__init__.py

```diff
@@ -333,3 +338,22 @@ def _module_name(path_list):
 
 def _module_to_cmd(module):
     return module.replace("_", "-")
+
+
+def _fix_sys_path():
+    """Remove the script directory from `sys.path`
+
+    The script's directory is added to sys.path by the interpreter. In
+    Python 3.11, this will change with the ``-P`` flag.
+    """
+    global _fix_sys_path_done
+
+    if _fix_sys_path_done:
+        return
+    _fix_sys_path_done = True
+    if not (sys.argv and sys.path):
+        # Not enough information
+        return
+    d = os.path.dirname(os.path.realpath(sys.argv[0]))
```

Comment:
Any reason for this check instead of `if sys.path[0] != ''; sys.path.pop(0)`

https://docs.python.org/3.7/library/sys.html#sys.path

Reply:
That's a different case which does not exist if pkcli.main is used. We always have a script. This case is really not an issue:

```sh
$ python
> import pykern.pkcli
> pykern.pkcli.main('pykern')
```

We also don't have the '-m' case, which is when you pass in a module.